### PR TITLE
fix: coerce partially decoded instructions on parsed confirmed transactions

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
PartiallyDecodedInnerInstructions contain a programId and accounts, which are all supposed to be PublicKey or PublicKey[] types. These are being provided as strings.

#### Summary of Changes
Coerce public keys on parsed confirmed transactions. 

Fixes https://github.com/solana-labs/solana/issues/13965

### CODE REVIEW

File: web3.js/src/connection.js [hunk 1]
```diff
@@ -368,14 +368,9 @@ const SimulatedTransactionResponseValidator = jsonRpcResultAndContext(
...
 
-type PartiallyDecodedInnerInstruction = {
-  index: number,
-  instructions: PartiallyDecodedInstruction[],
-};
-
 type ParsedInnerInstruction = {
   index: number,
-  instructions: (ParsedInstruction | PartiallyDecodedInnerInstruction)[],
+  instructions: (ParsedInstruction | PartiallyDecodedInstruction)[],
```
Comment (oJshua) [inline web3.js/src/connection.js:373]: This appeared to be incorrectly typed. Maybe someone can confirm.

File: web3.js/test/connection.test.js [hunk 2]
```diff
@@ -999,6 +999,117 @@ test('get confirmed transaction', async () => {
...
+      accounts: [
+        'EeJqWk5pczNjsqqY3jia9xfFNG1dD68te4s8gsdCuEk7',
+        '6tVrjJhFm5SAvvdh6tysjotQurCSELpxuW3JaAAYeC1m',
+      ],
+      data: 'ai3535',
+      programId: 'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
+    }),
+  ]);
+
+  //$FlowFixMe
```
Comment (oJshua) [inline web3.js/test/connection.test.js:1102]: I had a heck of a time getting flow type to work with this...

File: web3.js/src/connection.js (hunk 1 above)
Comment (jstarry) [reply web3.js/src/connection.js:373]: Your change is correct, thanks for fixing!

File: web3.js/test/connection.test.js (hunk 2 above)
Comment (jstarry) [reply web3.js/test/connection.test.js:1102]: Yeah, it's doable but not really worth the effort. This is fine

**Review by jstarry — APPROVED** [review]

**Pull request merged**